### PR TITLE
Add the ETag to the client response on the GET endpoints

### PR DESCRIPTION
**Pull request by rahmhatre**
This PR will add the ETag in the headers for the GET endpoints of the template service i.e WebPush, Email and SMS.

### CODE REVIEW

File: Solutions/Marain.UserNotifications.Specs/Features/ManagementApiClient/NotificationTemplate.feature [hunk 1]
```diff
@@ -49,13 +49,18 @@ Scenario: Get a Web Push notification template
 		"""
 	When I use the client to send the notification template API a request to get a notification template with notification type 'marain.test.template4' and communication type 'WebPush'
 	Then the client response status code should be 'OK'
+	And the client response should contain a 'If-None-Match' header
```
Comment (jongeorge1) [inline Solutions/Marain.UserNotifications.Specs/Features/ManagementApiClient/NotificationTemplate.feature:52]: I don't quite understand this. `If-None-Match` is a *request* header (by definition) - you'd never expect to see it in a response.

If an etag is being returned in a response, it should be in the `Etag` header.

**Review by jongeorge1 — CHANGES_REQUESTED** [review]

File: Solutions/Marain.UserNotifications.Specs/Features/ManagementApiClient/NotificationTemplate.feature (hunk 1 above)
Comment (rahmhatre) [reply Solutions/Marain.UserNotifications.Specs/Features/ManagementApiClient/NotificationTemplate.feature:52]: Okay. Sorry, I might have got confused while reading that from the host response and passed that straight through. I shall fix that. Thanks @jongeorge1

Comment (rahmhatre) [reply Solutions/Marain.UserNotifications.Specs/Features/ManagementApiClient/NotificationTemplate.feature]: @jongeorge1 I have updated it to add the ETag header correctly. Thanks for that 😊

**Review by jongeorge1 — DISMISSED** [review]

File: Solutions/Marain.UserNotifications.Client/Marain/UserNotifications/Client/Management/UserNotificationsManagementClient.cs [hunk 2]
```diff
@@ -299,5 +309,16 @@ private HttpRequestMessage AddETagToNotificationTemplateHeaders(ICommunicationTe
 
             return null;
         }
+
+        private ImmutableDictionary<string, string> AddETagToImmutableDictionary(HttpResponseMessage response)
```
Comment (idg10) [inline Solutions/Marain.UserNotifications.Client/Marain/UserNotifications/Client/Management/UserNotificationsManagementClient.cs:313]: I think this method has two problems:

- it is misnamed
- it is an instance member but apparently does not need to be

This caused me to misunderstand what was happening at the point where it was called. Given that this is an instance method that claims to add something to a dictionary I assumed that meant there was already some dictionary somewhere associated with this object that this method would return a modified version of.

In fact, if I've understood it, this returns a brand new dictionary containing either a single item, the `ETag`, or containing nothing if there is no `ETag`. So maybe this should be a static method called something like `WrapETagInImmutableDictionary`?

**Review by idg10 — CHANGES_REQUESTED** [review]

File: Solutions/Marain.UserNotifications.Client/Marain/UserNotifications/Client/Management/UserNotificationsManagementClient.cs (hunk 2 above)
Comment (rahmhatre) [reply Solutions/Marain.UserNotifications.Client/Marain/UserNotifications/Client/Management/UserNotificationsManagementClient.cs]: Hi Ian, I have renamed this method to the one you have suggested and made it static.

**Review by idg10 — APPROVED** [review]

**Comment by jongeorge1** [comment]
Has now been merged via #137

**Pull request closed without merging**